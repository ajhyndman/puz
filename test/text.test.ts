import { readFileSync, writeFileSync } from 'fs';
import { join } from 'path';

import { parseTextFile, printBinaryFile, printTextFile } from '../src/index';

describe('puz', () => {
  const puzzleFiles = [
    '2023-12-25-christmas-cipher.txt',
    'v1_text_format.txt',
    'v2_text_format_mark.txt',
    'v2_text_format_rebus.txt',
  ].map((path) => [path, readFileSync(join(__dirname, 'puzzles', path))] as const);

  describe('parseTextFile', () => {
    it('throws if file signature is not present', () => {
      expect(() => parseTextFile('')).toThrowError(
        'File does not appear to be an Across Lite puzzle description',
      );
    });

    it('throws if not all required tags are present', () => {
      expect(() => parseTextFile('<ACROSS PUZZLE>')).toThrowError(
        'File is missing required tag: <TITLE>',
      );
      expect(() => parseTextFile('<ACROSS PUZZLE V2>')).toThrowError(
        'File is missing required tag: <TITLE>',
      );
      expect(() =>
        parseTextFile(`<ACROSS PUZZLE>
        <TITLE>
        <AUTHOR>
        <COPYRIGHT>
        <SIZE>
        <GRID>
        <DOWN>
      `),
      ).toThrowError('File is missing required tag: <ACROSS>');
      expect(() => parseTextFile('<ACROSS PUZZLE V2>')).toThrowError(
        'File is missing required tag: <TITLE>',
      );
    });

    it.each(puzzleFiles)('parses "%s" without crashing', (path, buffer) => {
      parseTextFile(buffer.toString('utf-8'));
    });

    it.each(puzzleFiles)('extracts expected Puzzle object from "%s"', (path, buffer) => {
      expect(parseTextFile(buffer.toString('utf-8'))).toMatchSnapshot();
    });
  });

  describe('printTextFiles', () => {
    it.each(puzzleFiles)('prints "%s" back to expected V2 file', (path, buffer) => {
      const puzzle = parseTextFile(buffer.toString('utf-8'));
      expect(printTextFile(puzzle)).toMatchSnapshot();
    });
  });

  describe('convert text file to binary', () => {
    const inPath = join(__dirname, 'puzzles', '2023-12-25-christmas-cipher.txt');
    const buffer = readFileSync(inPath);
    const puzzle = parseTextFile(buffer.toString('utf-8'));
    let binaryFile: Uint8Array;
    binaryFile = printBinaryFile(puzzle);
    const outPath = join(__dirname, 'puzzles', '2023-12-25-christmas-cipher.puz');
    writeFileSync(outPath, binaryFile);
  });
});
