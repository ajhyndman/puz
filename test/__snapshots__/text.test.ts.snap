// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`puz parseTextFile extracts expected Puzzle object from "2023-12-25-christmas-cipher.txt" 1`] = `
Object {
  "author": "The circled letters reveal the secret!",
  "clues": Array [
    "Vehicle model named for the Atlantic Jetstream",
    "Brother of Esau",
    "It came before Slack",
    "A collection of figures",
    "A brief space of time",
    "Role-playing with Wynken and Blynken?",
    "___ day is equal to 24 hours, 37 minutes and 22 seconds",
    "It might be made of logs",
    "One who might sell petroleum",
    "Leak, to Murderbot",
  ],
  "copyright": "©2023 Andrew Hyndman",
  "height": 5,
  "markupGrid": Array [
    Object {
      "circled": true,
    },
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {
      "circled": true,
    },
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {
      "circled": true,
    },
    Object {},
    Object {},
    Object {},
    Object {},
    Object {
      "circled": true,
    },
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {
      "circled": true,
    },
  ],
  "notepad": "  The circled letters reveal the secret!",
  "solution": "JETTAAMARSCABINOILCOBLEED",
  "title": "December 25, 2023 — Christmas Cipher",
  "width": 5,
}
`;

exports[`puz parseTextFile extracts expected Puzzle object from "v1_text_format.txt" 1`] = `
Object {
  "author": "Created by Avalonian",
  "clues": Array [
    "Destiny",
    "Biting insect",
    "Troubles",
    "Golf equipments",
    "Computer keyboard key",
    "Above water, barely",
    "Oak seed or fruit",
    "Sausage",
    "I smell ____",
    "Squat",
    "Community dances",
    "Deserter",
    "Owned property",
    "House on Pennsylvania Ave.",
    "Seeps",
    "City NE of Manchester",
    "True ____ : Arnold S. movie",
    "Novel or rare item",
    "If the ____ fits, ...",
    "Favorite group of 53 across",
    "Subject of dentistry",
    "EEE with 16 across",
    "Midsummer's Night Dream character",
    "Formerly",
    "Ate sparingly",
    "Wife of Asiris",
    "Monetary unit",
    "Use reference",
    "She entertains guests",
    "\\"____, Johnny!\\"",
    "Make ready for use again",
    "South African currency unit",
    "Canal or Lake",
    "Delaying tactic of 53 across",
    "Yiddish, informally",
    "Spinning ____",
    "Math. subject",
    "Parts of typeset characters",
    "Toll",
    "Clod",
    "One who mimics",
    "Bearers : comb. form",
    "Egyptian river",
    "Bancroft and Archer",
    "Female pilot",
    "Fool",
    "Medics",
    "God in latin",
    "Pizza place fixtures",
    "Homages",
    "Sometimes a dieter is one",
    "Coat part",
    "Physical",
    "Idle",
    "Some kinds of snakes",
    "Type of sandwich",
    "Of posterior muscles",
    "Clean and care for",
    "\\"____ through!\\"",
    "Raises one's spirits",
    "Smallest planet of the Sun",
    "Block",
    "Swing about",
    "Cover",
    "____ slopes",
    "One who works despite a strike",
    "Glacial ridge",
    "Org.",
    "A dog's name",
    "GOP member",
    "Before",
    "Tax break savings account",
    "Of masses",
    "Angry",
    "Strikes violently",
    "Just",
    "Inert gas",
    "Encourage",
  ],
  "copyright": "1995 Literate Software Systems",
  "height": 15,
  "notepad": "This is an example notepad entry with
two lines in it.",
  "solution": "FATE.AWASH.AWOLLIES.CURIO.SHOEELECTORATE.SIZEASS.ERST.DIETED...CENT.HOSTESSREFITS.JEWISH..ARITH.KERNS.OAFNILE.ANNES.DUPEDEI.OVENS.LOSER..BODILY.RACERSGLUTEAL.PEPS...RESIST.SLUE.SKIOTTO.REPUBLICANOMES.IRATE.RAMSMERE.XENON.ABET",
  "title": "Politics: Who, what, where and why",
  "width": 15,
}
`;

exports[`puz parseTextFile extracts expected Puzzle object from "v2_text_format_mark.txt" 1`] = `
Object {
  "author": "Created by Avalonian",
  "clues": Array [
    "Destiny",
    "Biting insect",
    "Troubles",
    "Golf equipments",
    "Computer keyboard key",
    "Above water, barely",
    "Oak seed or fruit",
    "Sausage",
    "I smell ____",
    "Squat",
    "Community dances",
    "Deserter",
    "Owned property",
    "House on Pennsylvania Ave.",
    "Seeps",
    "City NE of Manchester",
    "True ____ : Arnold S. movie",
    "Novel or rare item",
    "If the ____ fits, ...",
    "Favorite group of 53 across",
    "Subject of dentistry",
    "EEE with 16 across",
    "Midsummer's Night Dream character",
    "Formerly",
    "Ate sparingly",
    "Wife of Asiris",
    "Monetary unit",
    "Use reference",
    "She entertains guests",
    "\\"____, Johnny!\\"",
    "Make ready for use again",
    "South African currency unit",
    "Canal or Lake",
    "Delaying tactic of 53 across",
    "Yiddish, informally",
    "Spinning ____",
    "Math. subject",
    "Parts of typeset characters",
    "Toll",
    "Clod",
    "One who mimics",
    "Bearers : comb. form",
    "Egyptian river",
    "Bancroft and Archer",
    "Female pilot",
    "Fool",
    "Medics",
    "God in latin",
    "Pizza place fixtures",
    "Homages",
    "Sometimes a dieter is one",
    "Coat part",
    "Physical",
    "Idle",
    "Some kinds of snakes",
    "Type of sandwich",
    "Of posterior muscles",
    "Clean and care for",
    "\\"____ through!\\"",
    "Raises one's spirits",
    "Smallest planet of the Sun",
    "Block",
    "Swing about",
    "Cover",
    "____ slopes",
    "One who works despite a strike",
    "Glacial ridge",
    "Org.",
    "A dog's name",
    "GOP member",
    "Before",
    "Tax break savings account",
    "Of masses",
    "Angry",
    "Strikes violently",
    "Just",
    "Inert gas",
    "Encourage",
  ],
  "copyright": "1995 Literate Software Systems",
  "height": 15,
  "markupGrid": Array [
    Object {
      "circled": true,
    },
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {
      "circled": true,
    },
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {
      "circled": true,
    },
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {},
    Object {
      "circled": true,
    },
  ],
  "notepad": "This is an example notepad entry with
two lines in it.",
  "solution": "FATE.AWASH.AWOLLIES.CURIO.SHOEELECTORATE.SIZEASS.ERST.DIETED...CENT.HOSTESSREFITS.JEWISH..ARITH.KERNS.OAFNILE.ANNES.DUPEDEI.OVENS.LOSER..BODILY.RACERSGLUTEAL.PEPS...RESIST.SLUE.SKIOTTO.REPUBLICANOMES.IRATE.RAMSMERE.XENON.ABET",
  "title": "Politics: Who, what, where and why",
  "width": 15,
}
`;

exports[`puz parseTextFile extracts expected Puzzle object from "v2_text_format_rebus.txt" 1`] = `
Object {
  "author": "Jeremy Newton / Will Shortz",
  "clues": Array [
    "Portraitist of George Washington",
    "Weightlifter's helper",
    "Have a break at 4:00, say",
    "Troop troupe: Abbr.",
    "Basic travel path",
    "First name in gossip",
    "Paris was part of it",
    "Robot maid on \\"The Jetsons\\"",
    "Equips with new clips",
    "Small African antelope",
    "It's darn likely",
    "\\"My, my, old chap!\\"",
    "Dusk, to Donne",
    "Sorority chapter",
    "Unlikely event for puritans",
    "MADD member",
    "Foe of Spider-Man",
    "Like some modern maps",
    "Coffee table item",
    "Leader of a flock",
    "Do away with",
    "Spreader of holiday cheer",
    "Mustang rivals",
    "Gene who sang \\"Back in the Saddle Again\\"",
    "Singer who said \\"At least I had that, one guy understood me\\"",
    "Cause of many uprisings",
    "Possible punishment for steroid use",
    "Enhance",
    "When Cannes heats up",
    "Native tongue of R&B singer Rihanna",
    "Bow to",
    "Wise guy",
    "Members of the bar?",
    "Ear part",
    "\\"Esq.\\" titleholders",
    "Biker's add-on",
    "Keep an eye out for",
    "Early Chinese dynasty",
    "Meager",
    "Admit defeat, in a way",
    "Prefix with sphere",
    "Lucratively",
    "Fuming",
    "Polite disclaimer",
    "ABC a.m. show, briefly",
    "Daring",
    "Last resort in poker, often",
    "Alvin and the Chipmunks, e.g.",
    "Experts at exports",
    "Doesn't waver",
    "Euripides drama",
    "Genre explored by Run-D.M.C. and Aerosmith",
    "G.P.S. suggestion",
    "Diurnally",
    "\\"Hop ___!\\"",
    "Mexican mouse chaser",
    "Bottle opener's surprise",
    "Vegas openings?",
    "Inner circle",
    "Untangles, in a way",
    "Cyberball maker",
    "Pop singer who appeared in the movie \\"Sgt. Pepper's Lonely Hearts Club Band\\"",
    "Proust title character",
    "It's called in a political convention roll call",
    "Undercover device",
    "Job interview topic",
    "Campaigns for",
    "\\"Quite possibly\\"",
    "Surface-___",
    "Singles and jingles",
    "Sioux tribe member",
    "State capital on the Colorado River",
    "Sub with sauerkraut?",
    "White wine from Verona",
    "Economy-size",
    "W., e.g.: Abbr.",
    "Langston Hughes poem",
    "Cry at sea",
    "Professional with many contacts?",
    "Cancún resident, once",
    "Lords and ladies",
    "Farm call",
    "Hall's partner in pop",
    "\\"This is how it's done\\"",
    "Worth mentioning",
    "Pattern for light or sound",
    "Is past?",
    "Soft hat materials",
    "Scrape (out)",
    "Actors Max and Max Jr.",
    "Like some professors",
    "See",
    "___ Taylor, co-host of \\"Make Me a Supermodel\\"",
    "Sask. neighbor",
    "It appears when things go bad",
    "Drinks in frosted glasses",
    "Married",
    "Boils down",
    "Hit Sony product introduced in 1984",
    "Model after",
    "American, Swiss, etc.",
    "Thrill seeker",
    "Design feature of many a viaduct",
    "Lookout, maybe",
    "\\"OMG, that's sooo funny!\\"",
    "Kind of bar",
    "Squared away",
    "First name in cosmetics",
    "\\"If you ask me,\\" online",
    "The \\"A\\" of James A. Garfield",
    "Now, in Nogales",
    "Wages, before overtime",
    "Some exams for joint pain sufferers",
    "Litigant",
    "Symptom of catarrh",
    "Focused (on)",
    "Beer serving in a pub",
    "\\"Yippee!\\"",
    "Brisk pace",
    "Cavs, on a scoreboard",
    "Nativity figure",
    "Jabbers",
    "\\"Was it ___ I saw?\\" (classic palindrome)",
    "U.S. gas chain",
    "Samuel L. Jackson's character in \\"Pulp Fiction\\"",
    "Triple-edged sword",
    "Part of a horse's genealogy",
    "Were present?",
    "It might run in the rain",
    "Cool ___",
    "Filled (with)",
    "Genealogical grp.",
    "Part of a serial",
    "S, on a French compass",
    "Eroded",
    "It's usually said with the eyes closed",
    "Wig",
    "Egg holders",
    "Kind of skill",
    "Farm machine",
  ],
  "copyright": "2008, The New York Times",
  "height": 21,
  "rebus": Object {
    "grid": Array [
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      1,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      0,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      2,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      12,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      3,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      11,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      4,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      9,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      5,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      8,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      6,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      7,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
      ,
    ],
    "solution": Object {
      "0": "JAN",
      "1": "FEB",
      "11": "NOV",
      "12": "DEC",
      "2": "MAR",
      "3": "APR",
      "4": "MAY",
      "5": "JUN",
      "6": "JUL",
      "7": "AUG",
      "8": "SEP",
      "9": "OCT",
    },
  },
  "solution": ".STUART.ROSIE...OMEGA.PASTOR.ERASE.CAROLERYOKOONO.LIFAN.AUGMENTETE.BAJ.OBEY.SMTY.COBATTS..AWAIT.POOR.ATMOSEEINGRED..GUTSY.TRIOTRADOMMISSIONS..RAOCK...ADAY..TOIT.GATO...CADRE..SWANN.BENEFITSOTO.AUSTIN.GIANT.ITOOMANERBOARD.ANCIENTMANBRAY.OATES.LIKESO.BIGSINEWAVE.FELTS..BAERS...DATE.NIKI..MOLD...DECOS..ADRENALINEJKIEIMHO.ABRAM..BONESCANSSUER.HACK.DWELT..TROTCLE.JOSH.ARAT.JES.AREMASCARA.FRAHT.EPISODEATEAWAY.ACHOO.PERUKE.NESTS...MOTOR.SEEDER.",
  "title": "NY Times, Sun, Sep 14, 2008  YEAR-ROUND",
  "width": 21,
}
`;

exports[`puz printTextFiles prints "2023-12-25-christmas-cipher.txt" back to expected V2 file 1`] = `
"<ACROSS PUZZLE V2>
<TITLE>
	December 25, 2023 — Christmas Cipher
<AUTHOR>
	The circled letters reveal the secret!
<COPYRIGHT>
	©2023 Andrew Hyndman
<SIZE>
	5x5
<GRID>
	jETTA
	AmARS
	CAbIN
	OIlCO
	BLEEd
<REBUS>
	MARK;
<ACROSS>
	Vehicle model named for the Atlantic Jetstream
	___ day is equal to 24 hours, 37 minutes and 22 seconds
	It might be made of logs
	One who might sell petroleum
	Leak, to Murderbot
<DOWN>
	Brother of Esau
	It came before Slack
	A collection of figures
	A brief space of time
	Role-playing with Wynken and Blynken?
<NOTEPAD>
  The circled letters reveal the secret!
"
`;

exports[`puz printTextFiles prints "v1_text_format.txt" back to expected V2 file 1`] = `
"<ACROSS PUZZLE V2>
<TITLE>
	Politics: Who, what, where and why
<AUTHOR>
	Created by Avalonian
<COPYRIGHT>
	1995 Literate Software Systems
<SIZE>
	15x15
<GRID>
	FATE.AWASH.AWOL
	LIES.CURIO.SHOE
	ELECTORATE.SIZE
	ASS.ERST.DIETED
	...CENT.HOSTESS
	REFITS.JEWISH..
	ARITH.KERNS.OAF
	NILE.ANNES.DUPE
	DEI.OVENS.LOSER
	..BODILY.RACERS
	GLUTEAL.PEPS...
	RESIST.SLUE.SKI
	OTTO.REPUBLICAN
	OMES.IRATE.RAMS
	MERE.XENON.ABET
<ACROSS>
	Destiny
	Above water, barely
	Deserter
	True ____ : Arnold S. movie
	Novel or rare item
	If the ____ fits, ...
	Favorite group of 53 across
	EEE with 16 across
	Midsummer's Night Dream character
	Formerly
	Ate sparingly
	Monetary unit
	She entertains guests
	Make ready for use again
	Yiddish, informally
	Math. subject
	Parts of typeset characters
	Clod
	Egyptian river
	Bancroft and Archer
	Fool
	God in latin
	Pizza place fixtures
	Sometimes a dieter is one
	Physical
	Some kinds of snakes
	Of posterior muscles
	Raises one's spirits
	Block
	Swing about
	____ slopes
	A dog's name
	GOP member
	Of masses
	Angry
	Strikes violently
	Just
	Inert gas
	Encourage
<DOWN>
	Biting insect
	Troubles
	Golf equipments
	Computer keyboard key
	Oak seed or fruit
	Sausage
	I smell ____
	Squat
	Community dances
	Owned property
	House on Pennsylvania Ave.
	Seeps
	City NE of Manchester
	Subject of dentistry
	Wife of Asiris
	Use reference
	\\"____, Johnny!\\"
	South African currency unit
	Canal or Lake
	Delaying tactic of 53 across
	Spinning ____
	Toll
	One who mimics
	Bearers : comb. form
	Female pilot
	Medics
	Homages
	Coat part
	Idle
	Type of sandwich
	Clean and care for
	\\"____ through!\\"
	Smallest planet of the Sun
	Cover
	One who works despite a strike
	Glacial ridge
	Org.
	Before
	Tax break savings account
<NOTEPAD>
This is an example notepad entry with
two lines in it.
"
`;

exports[`puz printTextFiles prints "v2_text_format_mark.txt" back to expected V2 file 1`] = `
"<ACROSS PUZZLE V2>
<TITLE>
	Politics: Who, what, where and why
<AUTHOR>
	Created by Avalonian
<COPYRIGHT>
	1995 Literate Software Systems
<SIZE>
	15x15
<GRID>
	fATE.AWASH.AWOl
	LIES.CURIO.SHOE
	ELECTORATE.SIZE
	ASS.ERST.DIETED
	...CENT.HOSTESS
	REFITS.JEWISH..
	ARITH.KERNS.OAF
	NILE.ANNES.DUPE
	DEI.OVENS.LOSER
	..BODILY.RACERS
	GLUTEAL.PEPS...
	RESIST.SLUE.SKI
	OTTO.REPUBLICAN
	OMES.IRATE.RAMS
	mERE.XENON.ABEt
<REBUS>
	MARK;
<ACROSS>
	Destiny
	Above water, barely
	Deserter
	True ____ : Arnold S. movie
	Novel or rare item
	If the ____ fits, ...
	Favorite group of 53 across
	EEE with 16 across
	Midsummer's Night Dream character
	Formerly
	Ate sparingly
	Monetary unit
	She entertains guests
	Make ready for use again
	Yiddish, informally
	Math. subject
	Parts of typeset characters
	Clod
	Egyptian river
	Bancroft and Archer
	Fool
	God in latin
	Pizza place fixtures
	Sometimes a dieter is one
	Physical
	Some kinds of snakes
	Of posterior muscles
	Raises one's spirits
	Block
	Swing about
	____ slopes
	A dog's name
	GOP member
	Of masses
	Angry
	Strikes violently
	Just
	Inert gas
	Encourage
<DOWN>
	Biting insect
	Troubles
	Golf equipments
	Computer keyboard key
	Oak seed or fruit
	Sausage
	I smell ____
	Squat
	Community dances
	Owned property
	House on Pennsylvania Ave.
	Seeps
	City NE of Manchester
	Subject of dentistry
	Wife of Asiris
	Use reference
	\\"____, Johnny!\\"
	South African currency unit
	Canal or Lake
	Delaying tactic of 53 across
	Spinning ____
	Toll
	One who mimics
	Bearers : comb. form
	Female pilot
	Medics
	Homages
	Coat part
	Idle
	Type of sandwich
	Clean and care for
	\\"____ through!\\"
	Smallest planet of the Sun
	Cover
	One who works despite a strike
	Glacial ridge
	Org.
	Before
	Tax break savings account
<NOTEPAD>
This is an example notepad entry with
two lines in it.
"
`;

exports[`puz printTextFiles prints "v2_text_format_rebus.txt" back to expected V2 file 1`] = `
"<ACROSS PUZZLE V2>
<TITLE>
	NY Times, Sun, Sep 14, 2008  YEAR-ROUND
<AUTHOR>
	Jeremy Newton / Will Shortz
<COPYRIGHT>
	2008, The New York Times
<SIZE>
	21x21
<GRID>
	.STUART.ROSIE...OMEGA
	.PASTOR.ERASE.CAROLER
	YOKOONO.LI2AN.AUGMENT
	ETE.BA1.OBEY.S3TY.COB
	ATTS..AWAIT.POOR.ATMO
	SEEINGRED..GUTSY.TRIO
	TRA#OMMISSIONS..R4OCK
	...ADAY..TOIT.GATO...
	CADRE..SWANN.BENEFITS
	OTO.AUSTIN.GIANT.ITOO
	MA@ERBOARD.ANCIENT5AN
	BRAY.OATES.LIKESO.BIG
	SINEWAVE.FELTS..BAERS
	...DATE.NIKI..MOLD...
	DEC0S..ADRENALINE6KIE
	IMHO.ABRAM..BONESCANS
	SUER.HACK.DWELT..TROT
	CLE.JO9H.ARAT.7ES.ARE
	MASCARA.FR8HT.EPISODE
	ATEAWAY.ACHOO.PERUKE.
	NESTS...MOTOR.SEEDER.
<REBUS>
	1:JAN:J
	2:FEB:F
	3:MAR:M
	4:APR:A
	5:MAY:M
	6:JUN:J
	7:JUL:J
	8:AUG:A
	9:SEP:S
	0:OCT:O
	@:NOV:N
	#:DEC:D
<ACROSS>
	Portraitist of George Washington
	Robot maid on \\"The Jetsons\\"
	Sorority chapter
	Leader of a flock
	Do away with
	Spreader of holiday cheer
	Singer who said \\"At least I had that, one guy understood me\\"
	Possible punishment for steroid use
	Enhance
	When Cannes heats up
	Native tongue of R&B singer Rihanna
	Bow to
	Wise guy
	Ear part
	\\"Esq.\\" titleholders
	Keep an eye out for
	Meager
	Prefix with sphere
	Fuming
	Daring
	Alvin and the Chipmunks, e.g.
	Experts at exports
	Genre explored by Run-D.M.C. and Aerosmith
	Diurnally
	\\"Hop ___!\\"
	Mexican mouse chaser
	Inner circle
	Proust title character
	Job interview topic
	Sioux tribe member
	State capital on the Colorado River
	Economy-size
	Langston Hughes poem
	Cry at sea
	Cancún resident, once
	Farm call
	Hall's partner in pop
	\\"This is how it's done\\"
	Worth mentioning
	Pattern for light or sound
	Soft hat materials
	Actors Max and Max Jr.
	See
	___ Taylor, co-host of \\"Make Me a Supermodel\\"
	It appears when things go bad
	Boils down
	Thrill seeker
	\\"If you ask me,\\" online
	The \\"A\\" of James A. Garfield
	Some exams for joint pain sufferers
	Litigant
	Symptom of catarrh
	Focused (on)
	Brisk pace
	Cavs, on a scoreboard
	Nativity figure
	\\"Was it ___ I saw?\\" (classic palindrome)
	Samuel L. Jackson's character in \\"Pulp Fiction\\"
	Were present?
	It might run in the rain
	Filled (with)
	Part of a serial
	Eroded
	It's usually said with the eyes closed
	Wig
	Egg holders
	Kind of skill
	Farm machine
<DOWN>
	Weightlifter's helper
	Have a break at 4:00, say
	Troop troupe: Abbr.
	Basic travel path
	First name in gossip
	Paris was part of it
	Equips with new clips
	Small African antelope
	It's darn likely
	\\"My, my, old chap!\\"
	Dusk, to Donne
	Unlikely event for puritans
	MADD member
	Foe of Spider-Man
	Like some modern maps
	Coffee table item
	Mustang rivals
	Gene who sang \\"Back in the Saddle Again\\"
	Cause of many uprisings
	Members of the bar?
	Biker's add-on
	Early Chinese dynasty
	Admit defeat, in a way
	Lucratively
	Polite disclaimer
	ABC a.m. show, briefly
	Last resort in poker, often
	Doesn't waver
	Euripides drama
	G.P.S. suggestion
	Bottle opener's surprise
	Vegas openings?
	Untangles, in a way
	Cyberball maker
	Pop singer who appeared in the movie \\"Sgt. Pepper's Lonely Hearts Club Band\\"
	It's called in a political convention roll call
	Undercover device
	Campaigns for
	\\"Quite possibly\\"
	Surface-___
	Singles and jingles
	Sub with sauerkraut?
	White wine from Verona
	W., e.g.: Abbr.
	Professional with many contacts?
	Lords and ladies
	Is past?
	Scrape (out)
	Like some professors
	Sask. neighbor
	Drinks in frosted glasses
	Married
	Hit Sony product introduced in 1984
	Model after
	American, Swiss, etc.
	Design feature of many a viaduct
	Lookout, maybe
	\\"OMG, that's sooo funny!\\"
	Kind of bar
	Squared away
	First name in cosmetics
	Now, in Nogales
	Wages, before overtime
	Beer serving in a pub
	\\"Yippee!\\"
	Jabbers
	U.S. gas chain
	Triple-edged sword
	Part of a horse's genealogy
	Cool ___
	Genealogical grp.
	S, on a French compass
"
`;
